discovery_engine_core/web-api/src/models.rs

```diff
@@ -16,11 +16,19 @@ use chrono::{NaiveDateTime, Utc};
 use derive_more::Display;
 use serde::{Deserialize, Serialize};
 use std::{collections::HashMap, str::FromStr};
+use thiserror::Error;
 use uuid::Uuid;
 
 use xayn_discovery_engine_ai::{Document as AiDocument, DocumentId, Embedding};
 use xayn_discovery_engine_core::document::Id;
 
+/// Web API errors.
+#[derive(Error, Debug)]
+pub(crate) enum Error {
+    #[error("`UserId` can't be empty")]
```

Comment:
:art: similarly to the other error types, we can `#[derive(displaydoc::Display)]` and use the doc comment for the error variant:
```suggestion
    /// `UserId` can't be empty
```